src/features/image/image.tsx

```diff
@@ -0,0 +1,52 @@
+/* eslint-disable @typescript-eslint/ban-ts-comment */
+import { FC, useState } from 'react';
+import { IS_DEV } from '@/features/image/constants.ts';
+import {
+  DecodingAttr,
+  FetchPriorityAttr,
+  ImageProps,
+  LoadingAttr,
+} from '@/features/image/types.ts';
+import convertToWebp from '@/features/image/utils/convertToWebp.ts';
+import generateSizes from '@/features/image/utils/generateSizes.ts';
+import generateSrcSet from '@/features/image/utils/generateSrcSet.ts';
+
+const Image: FC<ImageProps> = ({ alt, src = '', lazy = 'true', ...props }) => {
+  const srcWebp = convertToWebp(src);
+  const [srcSet, setSrcSet] = useState(() => (IS_DEV ? undefined : generateSrcSet(srcWebp)));
+  const [sizes, setSizes] = useState(() => (IS_DEV ? undefined : generateSizes()));
+
+  const isLazy = lazy === 'true';
+  const loading: LoadingAttr = isLazy ? 'lazy' : 'eager';
+  const fetchPriority: FetchPriorityAttr = isLazy ? 'low' : 'high';
+  const decoding: DecodingAttr = isLazy ? 'async' : 'auto';
+  const srcAttr = IS_DEV ? src : srcWebp;
+
+  const handleError = () => {
+    // fallback to basic src if there are no responsive sizes for an image
```

Comment:
In general comments should be avoided at all costs (although here they can be justified).
Usually prefer using meaningful names instead of a comment.

Reply:
I 100% agree with meaningful names instead of a comment, but i'm wondering why comments should be avoided at all costs? They can explain some tricky moments for some future developers

Reply:
https://www.codeproject.com/Articles/872073/Code-Comments-are-Lies